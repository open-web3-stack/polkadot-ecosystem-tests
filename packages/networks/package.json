{
	"name": "@e2e-test/networks",
	"type": "module",
	"dependencies": {
		"@acala-network/chopsticks-testing": "^0.13.1",
		"@e2e-test/shared": "workspace:*"
	},
	"main": "./dist/cjs/index.js",
	"module": "./dist/esm/index.js",
	"files": [
		"*",
		"!tsconfig.json",
		"!**/*.test.ts"
	],
	"exports": {
		".": {
			"require": "./dist/cjs/index.js",
			"import": "./dist/esm/index.js",
			"default": "./dist/esm/index.js"
		},
		"./chains": {
			"require": "./dist/cjs/chains/index.js",
			"import": "./dist/esm/chains/index.js",
			"default": "./dist/esm/chains/index.js"
		},
		"./*": {
			"require": "./dist/cjs/*.js",
			"import": "./dist/esm/*.js",
			"default": "./dist/esm/*.js"
		}
	}
}
