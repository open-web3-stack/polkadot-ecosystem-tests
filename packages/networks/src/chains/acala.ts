import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  acala: {
    relayToken: 'DOT',
    relayLiquidToken: 'LDOT',
    stableToken: 'AUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    dot: { Token: 'DOT' },
    ldot: { Token: 'LDOT' },
    dai: { Erc20: '0x54a37a01cd75b616d63e0ab665bffdb0143c52ae' },
    wbtc: { ForeignAsset: 5 },
    ausd: { Token: 'AUSD' },
    aca: { Token: 'ACA' },
    lcdot: { LiquidCrowdloan: 13 },
  },
  karura: {
    relayToken: 'KSM',
    relayLiquidToken: 'LKSM',
    stableToken: 'KUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    ksm: { Token: 'KSM' },
    lksm: { Token: 'LKSM' },
    usdt: { ForeignAsset: 7 },
    rmrk: { ForeignAsset: 0 },
    dai: { Erc20: '0x4bb6afb5fa2b07a5d1c499e1c3ddb5a15e709a71' },
    ausd: { Token: 'KUSD' },
    kar: { Token: 'KAR' },
  },
}

const getInitStorages = (config: typeof custom.acala | typeof custom.karura) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 4, data: { free: 10 * 1e12 } }]],
  },
  Tokens: {
    accounts: [
      [[defaultAccount.alice.address, { Token: config.relayToken }], { free: 10 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.relayLiquidToken }], { free: 100 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.stableToken }], { free: 1000 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccount.alice.address,
  },
  EvmAccounts: {
    accounts: [[['0x82a258cb20e2adb4788153cd5eb5839615ece9a0'], defaultAccount.alice.address]],
    evmAddresses: [[[defaultAccount.alice.address], '0x82a258cb20e2adb4788153cd5eb5839615ece9a0']],
  },
  Homa: {
    // avoid impact test outcome
    $removePrefix: ['redeemRequests', 'unbondings', 'toBondPool'],
    // so that bump era won't trigger unbond
    relayChainCurrentEra: 100,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const acala = defineChain({
  name: 'acala',
  endpoint: 'wss://acala-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.acala,
  initStorages: getInitStorages(custom.acala),
})

export const karura = defineChain({
  name: 'karura',
  endpoint: 'wss://karura-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.karura,
  initStorages: getInitStorages(custom.karura),
})
