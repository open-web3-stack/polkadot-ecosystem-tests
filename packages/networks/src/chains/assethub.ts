import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
  assetHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubPolkadot | typeof custom.assetHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubPolkadot = defineChain({
  name: 'assetHubPolkadot',
  endpoint: 'wss://rpc-asset-hub-polkadot.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubPolkadot,
  initStorages: getInitStorages(custom.assetHubPolkadot),
})

export const assetHubKusama = defineChain({
  name: 'assetHubKusama',
  endpoint: 'wss://rpc-asset-hub-kusama.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubKusama,
  initStorages: getInitStorages(custom.assetHubKusama),
})
