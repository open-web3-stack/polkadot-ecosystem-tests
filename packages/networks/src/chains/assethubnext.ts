import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNextWestend: {
    wnd: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNextWestend) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNextWestend = defineChain({
  name: 'assetHubNextWestend',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNextWestend,
  initStorages: getInitStorages(custom.assetHubNextWestend),
})
