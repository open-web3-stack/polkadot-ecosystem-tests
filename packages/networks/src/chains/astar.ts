import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  astar: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551617',

    paraAccount: '13YMK2eZzuFY1WZGagpYtTgbWBWGdoUD2CtrPj1mQPjY8Ldc',
    dot: 340282366920938463463374607431768211455n,
    astr: { Concrete: { parents: 0, interior: 'Here' } },
    aca: 18446744073709551616n,
    usdt: 4294969280n,
  },
  shiden: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551616',

    paraAccount: 'F7fq1jNy74AqkJ1DP4KqSrWtnTGtXfNVoDwFhTvvPxUvJaq',
    ksm: 340282366920938463463374607431768211455n,
    sdn: { Concrete: { parents: 0, interior: 'Here' } },
    kar: 18446744073709551618n,
    usdt: 4294969280n,
  },
}

const getInitStorages = (config: typeof custom.astar | typeof custom.shiden) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: '100000000000000000000' } }]],
  },
  Assets: {
    account: [
      [[config.relayToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
      [[config.aUSDToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccounts.alice.address,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const astar = defineChain({
  name: 'astar',
  paraId: 2006,
  endpoint: 'wss://astar-rpc.dwellir.com',
  custom: custom.astar,
  initStorages: getInitStorages(custom.astar),
})

export const shiden = defineChain({
  name: 'shiden',
  paraId: 2007,
  endpoint: 'wss://shiden-rpc.dwellir.com',
  custom: custom.shiden,
  initStorages: getInitStorages(custom.shiden),
})
