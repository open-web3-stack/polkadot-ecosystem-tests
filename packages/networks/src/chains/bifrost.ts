import { defineChain } from '../defineChain.js'
import { defaultAccounts, defaultAccountsSr25519, testAccounts } from '../testAccounts.js'

const custom = {
  bifrostPolkadot: {
    relayToken: 'DOT',
    bnc: { Native: 'BNC' },
  },
  bifrostKusama: {
    relayToken: 'KSM',
    bnc: { Token: 'BNC' },
  },
}

const getInitStorages = (_config: typeof custom.bifrostPolkadot | typeof custom.bifrostKusama) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[testAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
    ],
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers'],
  },
})

export const bifrostPolkadot = defineChain({
  name: 'bifrostPolkadot',
  endpoint: 'wss://bifrost-polkadot.ibp.network',
  paraId: 2030,
  custom: custom.bifrostPolkadot,
  initStorages: getInitStorages(custom.bifrostPolkadot),
})

export const bifrostKusama = defineChain({
  name: 'bifrostKusama',
  endpoint: 'wss://us.bifrost-rpc.liebi.com/ws',
  paraId: 2001,
  custom: custom.bifrostKusama,
  initStorages: getInitStorages(custom.bifrostKusama),
})
