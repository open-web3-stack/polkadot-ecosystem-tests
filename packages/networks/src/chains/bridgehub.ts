import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  bridgeHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
  },
  bridgeHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
  },
}

const getInitStorages = (_config: typeof custom.bridgeHubPolkadot | typeof custom.bridgeHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
})

export const bridgeHubPolkadot = defineChain({
  name: 'bridgeHubPolkadot',
  endpoint: 'wss://polkadot-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubPolkadot,
  initStorages: getInitStorages(custom.bridgeHubPolkadot),
})

export const bridgeHubKusama = defineChain({
  name: 'bridgeHubKusama',
  endpoint: 'wss://kusama-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubKusama,
  initStorages: getInitStorages(custom.bridgeHubKusama),
})
