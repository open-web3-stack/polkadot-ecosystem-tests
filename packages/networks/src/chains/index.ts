export * from './acala.js'
export * from './assethub.js'
export * from './astar.js'
export * from './hydraDX.js'
export * from './moonbeam.js'
export * from './polkadot.js'
