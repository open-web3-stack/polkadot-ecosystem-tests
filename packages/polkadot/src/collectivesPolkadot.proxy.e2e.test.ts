import { collectivesPolkadot } from '@e2e-test/networks/chains'

import { fullProxyE2ETests, registerTestTree } from '@e2e-test/shared'
import { CollectivesProxyTypes } from '@e2e-test/shared'

registerTestTree(
  fullProxyE2ETests(
    collectivesPolkadot,
    { testSuiteName: 'Polkadot Collectives Proxy', addressEncoding: 0 },
    CollectivesProxyTypes,
  ),
)
