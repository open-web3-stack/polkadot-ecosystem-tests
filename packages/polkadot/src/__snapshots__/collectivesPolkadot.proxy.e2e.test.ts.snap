// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Polkadot Collectives Proxy > add proxies (with/without delay) to an account, and remove them > events when adding proxies to Alice 1`] = `
[
  {
    "data": {
      "delay": 0,
      "delegatee": "1LBbHGCXcmzwxhmxsMvjGWfR6cGbUADPL5nv1A18rD4Xf4R",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "Any",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "14KRavukzykAQ7mWYDRWHnUAp42zpVeRgVr5Vu3oaAHBkn4J",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "NonTransfer",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "13foYqdELfj1VBWUdKpJTkLgr7tkGdSvxWyYbFjZaYi8iLan",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "CancelProxy",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "14QPRmts91hw16ReRoPSKLHdm1rEKVHftYcqEJyuWBwfnuJ6",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "Collator",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "13tsR5Cx9rYpR5UgpYskh3cyfiRxgVJiVvSJR89qLBjxtbA9",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "Alliance",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "12mR752ZLe32C5xhai4kE9n1sytSpE1XPTXr7dmzv2tYMMVs",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "Fellowship",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "14YRWSTsjvjY39pPXKu9fj1zHyso2E2mpxFmdtw6dMuyKjdW",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "Ambassador",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > add proxies (with/without delay) to an account, and remove them > events when removing all proxies from Alice 1`] = `[]`;

exports[`Polkadot Collectives Proxy > add proxies (with/without delay) to an account, and remove them > events when removing proxies from Alice (batch) 1`] = `
[
  {
    "data": {
      "delay": 0,
      "delegatee": "1LBbHGCXcmzwxhmxsMvjGWfR6cGbUADPL5nv1A18rD4Xf4R",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "Any",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "14KRavukzykAQ7mWYDRWHnUAp42zpVeRgVr5Vu3oaAHBkn4J",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "NonTransfer",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "13foYqdELfj1VBWUdKpJTkLgr7tkGdSvxWyYbFjZaYi8iLan",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "CancelProxy",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "14QPRmts91hw16ReRoPSKLHdm1rEKVHftYcqEJyuWBwfnuJ6",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "Collator",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "13tsR5Cx9rYpR5UgpYskh3cyfiRxgVJiVvSJR89qLBjxtbA9",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "Alliance",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "12mR752ZLe32C5xhai4kE9n1sytSpE1XPTXr7dmzv2tYMMVs",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "Fellowship",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "14YRWSTsjvjY39pPXKu9fj1zHyso2E2mpxFmdtw6dMuyKjdW",
      "delegator": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "proxyType": "Ambassador",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > create and kill pure proxies > events when creating pure proxies for Alice 1`] = `
[
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "Any",
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "NonTransfer",
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "CancelProxy",
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "Collator",
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "Alliance",
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "Fellowship",
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "Ambassador",
      "who": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > create and kill pure proxies > events when killing pure proxy of type 0 for Alice 1`] = `
[
  {
    "data": {
      "result": "Ok",
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > create and kill pure proxies > events when killing pure proxy of type 1 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > create and kill pure proxies > events when killing pure proxy of type 2 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > create and kill pure proxies > events when killing pure proxy of type 3 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > create and kill pure proxies > events when killing pure proxy of type 4 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > create and kill pure proxies > events when killing pure proxy of type 5 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > create and kill pure proxies > events when killing pure proxy of type 6 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > perform proxy call on behalf of delegator > events when Bob transfers funds to Charlie as Alice's proxy 1`] = `
[
  {
    "data": {
      "amount": 1000000000000,
      "from": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "to": "14Gjs1TD93gnwEBfDMHoCgsuf1s2TVKUP6Z1qKmAZnZ8cW5q",
    },
    "method": "Transfer",
    "section": "balances",
  },
  {
    "data": {
      "result": "Ok",
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > proxy announcement lifecycle test > events when Alice rejects Bob's proxy call announcement 1`] = `[]`;

exports[`Polkadot Collectives Proxy > proxy announcement lifecycle test > events when Bob announces a proxy call 1`] = `
[
  {
    "data": {
      "callHash": "(hash)",
      "proxy": "14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3",
      "real": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "method": "Announced",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > proxy announcement lifecycle test > events when Bob performs the announced proxy call 1`] = `
[
  {
    "data": {
      "result": "Ok",
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > proxy announcement lifecycle test > events when Bob removes their proxy call announcement 1`] = `[]`;

exports[`Polkadot Collectives Proxy > proxy call filtering test for Any > events for proxy action: proxy type Any, pallet balances, call transferKeepAlive 1`] = `
[
  {
    "data": {
      "who": "1LBbHGCXcmzwxhmxsMvjGWfR6cGbUADPL5nv1A18rD4Xf4R",
    },
    "method": "Upgraded",
    "section": "balances",
  },
  {
    "data": {
      "amount": "(rounded 14000000)",
      "who": "1LBbHGCXcmzwxhmxsMvjGWfR6cGbUADPL5nv1A18rD4Xf4R",
    },
    "method": "Withdraw",
    "section": "balances",
  },
  {
    "data": {
      "account": "16D2eVuK5SWfwvtFD3gVdBC2nc2BafK31BY6PrbZHBAGew7L",
      "freeBalance": 1000000000000,
    },
    "method": "Endowed",
    "section": "balances",
  },
  {
    "data": {
      "amount": 1000000000000,
      "from": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
      "to": "16D2eVuK5SWfwvtFD3gVdBC2nc2BafK31BY6PrbZHBAGew7L",
    },
    "method": "Transfer",
    "section": "balances",
  },
  {
    "data": {
      "result": "Ok",
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
  {
    "data": {
      "amount": "(rounded 14000000)",
      "who": "13UVJyLgBASGhE2ok3TvxUfaQBGUt88JCcdYjHvUhvQkFTTx",
    },
    "method": "Deposit",
    "section": "balances",
  },
]
`;

exports[`Polkadot Collectives Proxy > proxy call filtering test for Any > events for proxy action: proxy type Any, pallet system, call remark 1`] = `
[
  {
    "data": {
      "result": "Ok",
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
  {
    "data": {
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 4700)",
          "refTime": "(rounded 300000000)",
        },
      },
    },
    "method": "ExtrinsicSuccess",
    "section": "system",
  },
]
`;

exports[`Polkadot Collectives Proxy > proxy call filtering test for Any > events for proxy action: proxy type Any, pallet system, call remarkWithEvent 1`] = `
[
  {
    "data": {
      "hash_": "(hash)",
      "sender": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "method": "Remarked",
    "section": "system",
  },
  {
    "data": {
      "result": "Ok",
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
  {
    "data": {
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 4700)",
          "refTime": "(rounded 300000000)",
        },
      },
    },
    "method": "ExtrinsicSuccess",
    "section": "system",
  },
]
`;

exports[`Polkadot Collectives Proxy > proxy call filtering test for Any > events for proxy action: proxy type Any, pallet utility, call batch 1`] = `
[
  {
    "data": {},
    "method": "BatchCompleted",
    "section": "utility",
  },
  {
    "data": {
      "result": "Ok",
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > proxy call filtering test for Any > events for proxy action: proxy type Any, pallet utility, call forceBatch 1`] = `
[
  {
    "data": {},
    "method": "BatchCompleted",
    "section": "utility",
  },
  {
    "data": {
      "result": "Ok",
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > proxy call filtering test for CancelProxy > events for proxy action: proxy type CancelProxy, pallet proxy, call rejectAnnouncement 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x01000000",
            "index": 42,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Polkadot Collectives Proxy > proxy call filtering test for NonTransfer > events for proxy action: proxy type NonTransfer, pallet system, call remark 1`] = `
[
  {
    "data": {
      "result": "Ok",
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
  {
    "data": {
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 4700)",
          "refTime": "(rounded 300000000)",
        },
      },
    },
    "method": "ExtrinsicSuccess",
    "section": "system",
  },
]
`;

exports[`Polkadot Collectives Proxy > proxy call filtering test for NonTransfer > events for proxy action: proxy type NonTransfer, pallet system, call remarkWithEvent 1`] = `
[
  {
    "data": {
      "hash_": "(hash)",
      "sender": "15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5",
    },
    "method": "Remarked",
    "section": "system",
  },
  {
    "data": {
      "result": "Ok",
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
  {
    "data": {
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 4700)",
          "refTime": "(rounded 300000000)",
        },
      },
    },
    "method": "ExtrinsicSuccess",
    "section": "system",
  },
]
`;
